install/installer/pkg/config/v1/envvars.go

```diff
@@ -0,0 +1,418 @@
+					slice = reflect.Append(slice, reflect.ValueOf(strings.TrimSpace(i)))
+				}
+
+				valueField.Set(slice)
+			default:
+				valueField.SetString(envvar)
+			}
+		}
+	}
+}
+
+type dockerConfigJson struct {
+	Auths map[string]interface{} `json:"auths"`
+}
+
+// BuildFromEnvvars complete the configuration based on envvars - this may remove existing values
+// **IMPORTANT** this function does not guarantee validity of the generated configuration
+func (v version) BuildFromEnvvars(in interface{}) error {
+	cfg, ok := in.(*Config)
+	if !ok {
+		return config.ErrInvalidType
+	}
+
+	envvars := ConfigEnvvars{}
+	envvars.load()
+
+	log.Infof("Detected envvars: %+v", envvars)
+
+	cfg.Domain = envvars.Domain
+
+	if licenseName := envvars.LicenseName; licenseName != "" {
+		log.Infof("Setting license name: %s", licenseName)
+		cfg.License = &ObjectRef{
+			Kind: ObjectRefSecret,
+			Name: licenseName,
+		}
+	} else {
+		cfg.License = nil
+	}
+
+	if httpProxy := envvars.HTTPProxyName; httpProxy != "" {
+		log.Infof("Setting HTTP proxy name: %s", httpProxy)
+		cfg.HTTPProxy = &ObjectRef{
+			Kind: ObjectRefSecret,
+			Name: httpProxy,
+		}
+	} else {
+		cfg.HTTPProxy = nil
+	}
+
+	log.Info("Setting Open VSX URL")
+	cfg.OpenVSX = OpenVSX{
+		URL: func() string {
+			if v := envvars.OpenVSXUrl; v != "" {
+				return v
+			}
+			return defaultOpenVSXURL
+		}(),
+	}
+
+	log.Infof("DB incluster: %v", envvars.DBInClusterEnabled)
+
+	cfg.Database.InCluster = pointer.Bool(true)
+	cfg.Database.CloudSQL = nil
+	cfg.Database.External = nil
+
+	if !envvars.DBInClusterEnabled {
+		cfg.Database.InCluster = pointer.Bool(false)
+
+		if envvars.DBCloudSQLEnabled {
+			log.Info("Configuring CloudSQLProxy for database")
+			cfg.Database.CloudSQL = &DatabaseCloudSQL{
+				Instance: envvars.DBCloudSQLInstance,
+				ServiceAccount: ObjectRef{
+					Kind: ObjectRefSecret,
+					Name: envvars.DBCloudSQLServiceAccountName,
+				},
+			}
+		} else {
+			log.Info("Configuring external database")
+			cfg.Database.External = &DatabaseExternal{
+				Certificate: ObjectRef{
+					Kind: ObjectRefSecret,
+					Name: envvars.DBExternalCertificateName,
+				},
+			}
+		}
+	}
+
+	cfg.Repository = defaultRepositoryUrl
+	cfg.ImagePullSecrets = nil
+	cfg.DropImageRepo = nil
+	cfg.ContainerRegistry.PrivateBaseImageAllowList = []string{}
+
+	if envvars.LocalRegistryEnabled {
+		log.Info("Configuring mirrored container registry for airgapped installation")
+
+		cfg.Repository = envvars.LocalRegistryAddress
+		cfg.ImagePullSecrets = append(cfg.ImagePullSecrets, ObjectRef{
+			Kind: ObjectRefSecret,
+			Name: envvars.ImagePullSecretName,
+		})
+		cfg.DropImageRepo = pointer.Bool(true)
+
+		cfg.ContainerRegistry.PrivateBaseImageAllowList = append(cfg.ContainerRegistry.PrivateBaseImageAllowList, envvars.LocalRegistryHost)
+	}
+
+	if envvars.RegistryDockerConfigEnabled {
+		log.Info("Extracting servers from the custom registry authentication")
+		var registryDockerConfig dockerConfigJson
```

Comment:
Was Wondering that this isn't useful, but looking at https://github.com/gitpod-io/gitpod/pull/12986/files#diff-2ab933b419a78565fb696be3db9ac5a3fc6156ac1556a717c52b865b7f087070R79, It does make sense